libraries/helpers.rb

```diff
@@ -0,0 +1,19 @@
+module VarnishCookbook
+  # Helper methods to be used in multiple Varnish cookbook libraries.
+  module Helpers
+    def varnish_version
+      cmd = 'varnishd -V 2>&1'
+      cmd = Mixlib::ShellOut.new(cmd)
+      cmd.environment['HOME'] = ENV.fetch('HOME', '/root')
+
+      begin
+        cmd.run_command
+        Chef::Log.info "varnish version detected: #{cmd}"
+        Chef::Log.info "varnish version detected: #{cmd.stdout}"
+        return cmd.stdout.match(/varnish-([0-9])\./).captures[0]
+      rescue => ex
+        Chef::Log.warn "Unable to run varnishd to get version.\nMessage: #{ex.message}"
```

Comment:
Should this be a `fail "Unable to run varnishd to get version.\nMessage: #{ex.message}"` instead of a warning? It sounds more a fatal thing than a warning.